File: apps/admin/src/userMessages.ts
const technicalTerms = [
  "api",
  "banco",
  "database",
  "supabase",
  "rest",
  "prisma",
  "token",
  "jwt",
  "fetch",
  "network",
  "timeout",
  "econnrefused",
  "etimedout",
  "enotfound",
  "payload",
  "status",
  "500",
  "404",
  "403",
  "401",
];

export function userFacingError(error: unknown, fallback: string) {
  const message = error instanceof Error ? error.message.trim() : "";
  if (!message) return fallback;
  const normalized = message.toLowerCase();
  if (technicalTerms.some((term) => normalized.includes(term))) return fallback;
  return message;
}

File: apps/motoboy/app/src/main/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/UserMessages.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

private val technicalErrorTerms = listOf(
  "api",
  "backend",
  "database",
  "supabase",
  "rest",
  "token",
  "jwt",
  "http",
  "json",
  "timeout",
  "connectexception",
  "unknownhost",
  "econnrefused",
  "etimedout",
  "enotfound",
  "status",
  "500",
  "404",
  "403",
  "401",
)

fun courierFacingError(failure: Throwable, fallback: String): String {
  val message = failure.message?.trim().orEmpty()
  if (message.isBlank()) return fallback
  val normalized = message.lowercase()
  return if (technicalErrorTerms.any { normalized.contains(it) }) fallback else message
}
